### PR TITLE
[inductor] Support additional operators

### PR DESCRIPTION
**Pull request by jansel**
- isinf
- isnan
- all
- any
- hardswish_
- leaky_relu_
- silu_
- log1p
- masked_fill_
- hardtanh_

### CODE REVIEW

File: torchinductor/codecache.py [hunk 1]
```diff
@@ -271,11 +271,13 @@ def pointwise_heuristics(size_hints):
...
         return apply_triton_config(triton_config(size_hints, 1024))
     if len(size_hints) == 2:
+        if not config.triton.autotune:
+            return apply_triton_config(triton_config(size_hints, 64, 64))
         return autotune(
             [
                 triton_config(size_hints, 64, 64),
-                triton_config(size_hints, 8, 256),
-                triton_config(size_hints, 256, 8),
+                triton_config(size_hints, 4, 256),
```
Comment (desertfire) [inline torchinductor/codecache.py:279]: Out of curiosity, how did you come up with these numbers?

File: torchinductor/codegen/triton.py [hunk 2]
```diff
@@ -435,6 +442,8 @@ def reduction(self, name, dtype, reduction_type, index, value):
         )
         sizes = [f"{tree.prefix.upper()}BLOCK" for tree in self.range_trees]
         sizes[-1] = "1"
+        if reduction_type == "any":
+            reduction_type = "max"
```
Comment (desertfire) [inline torchinductor/codegen/triton.py:446]: Add a comment to explain this?

**Review by desertfire — APPROVED** [review]

File: torchinductor/codegen/triton.py (hunk 2 above)
Comment (jansel) [reply torchinductor/codegen/triton.py:446]: I can add one.  Triton doesn't have any().

File: torchinductor/decomposition.py [hunk 3]
```diff
@@ -10,27 +11,50 @@
 
 decompositions = get_decompositions(
     [
+        aten._adaptive_avg_pool2d_backward,
```
Comment (Chillee) [inline torchinductor/decomposition.py:14]: Some of these decompositions don't actually exist, right?

**Comment by jansel** [comment]
Correct

**Pull request merged**